src/Datadog.Trace/Agent/Api.cs

```diff
@@ -74,72 +74,41 @@ public async Task<bool> SendTracesAsync(Span[][] traces)
                     request.AddHeader(AgentHttpHeaderNames.ContainerId, _containerId);
                 }
 
-                IApiResponse response = null;
+                bool success = false;
+                Exception exception = null;
 
                 try
                 {
-                    try
-                    {
-                        _statsd?.AppendIncrementCount(TracerMetricNames.Api.Requests);
-                        response = await request.PostAsync(traces, _formatterResolver).ConfigureAwait(false);
-                    }
-                    catch
-                    {
-                        // count the exceptions thrown by the HttpClient,
-                        // not responses with 5xx status codes
-                        // (which cause EnsureSuccessStatusCode() to throw below)
-                        _statsd?.AppendIncrementCount(TracerMetricNames.Api.Errors);
-                        throw;
-                    }
-
-                    if (_statsd != null)
-                    {
-                        // don't bother creating the tags array if trace metrics are disabled
-                        string[] tags = { $"status:{response.StatusCode}" };
-
-                        // count every response, grouped by status code
-                        _statsd.AppendIncrementCount(TracerMetricNames.Api.Responses, tags: tags);
-                    }
-
-                    // Attempt a retry if the status code is not SUCCESS
-                    if (response.StatusCode < 200 || response.StatusCode > 300)
-                    {
-                        if (retryCount >= retryLimit)
-                        {
-                            // stop retrying
-                            Log.Error("An error occurred while sending traces to the agent at {Endpoint}", _tracesEndpoint);
-                            return false;
-                        }
-
-                        // retry
-                        await Task.Delay(sleepDuration).ConfigureAwait(false);
-                        retryCount++;
-                        sleepDuration *= 2;
-
-                        continue;
-                    }
+                    success = await SendTracesAsync(traces, request).ConfigureAwait(false);
                 }
                 catch (Exception ex)
                 {
 #if DEBUG
                     if (ex.InnerException is InvalidOperationException ioe)
                     {
-                        Log.Error("An error occurred while sending traces to the agent at {Endpoint}\n{Exception}", ex, _tracesEndpoint, ex.ToString());
+                        Log.Error(ex, "An error occurred while sending traces to the agent at {0}\n{1}", _tracesEndpoint, ex.ToString());
```

Comment:
Maybe `Environment.NewLine` instead of `\n`?

Reply:
Well actually we shouldn't even be concatenating the exception since we pass it using the dedicated overload. Gonna fix this